lib/utils/pyexec.c

```diff
@@ -50,90 +50,101 @@ int pyexec_system_exit = 0;
 STATIC bool repl_display_debugging_info = 0;
 #endif
 
-#define EXEC_FLAG_PRINT_EOF (1)
-#define EXEC_FLAG_ALLOW_DEBUGGING (2)
-#define EXEC_FLAG_IS_REPL (4)
-#define EXEC_FLAG_SOURCE_IS_RAW_CODE (8)
-#define EXEC_FLAG_SOURCE_IS_VSTR (16)
-#define EXEC_FLAG_SOURCE_IS_FILENAME (32)
-
-// parses, compiles and executes the code in the lexer
-// frees the lexer before returning
-// EXEC_FLAG_PRINT_EOF prints 2 EOF chars: 1 after normal output, 1 after exception output
-// EXEC_FLAG_ALLOW_DEBUGGING allows debugging info to be printed after executing the code
-// EXEC_FLAG_IS_REPL is used for REPL inputs (flag passed on to mp_compile)
-STATIC int parse_compile_execute(const void *source, mp_parse_input_kind_t input_kind, int exec_flags) {
+// Parses, compiles and executes the code in the lexer.
+// Frees the lexer before returning.
+// Returns 0 for success, non-zero for an error or SystemExit.
+// PYEXEC_FLAG_PRINT_EOF prints 2 EOF chars: 1 after normal output, 1 after exception output
+// PYEXEC_FLAG_ALLOW_DEBUGGING allows debugging info to be printed after executing the code
+// PYEXEC_FLAG_IS_REPL is used for REPL inputs (flag passed on to mp_compile)
```

Comment:
and PYEXEC_FLAG_SOURCE_IS_RAW_CODE, PYEXEC_FLAG_SOURCE_IS_VSTR, PYEXEC_FLAG_SOURCE_IS_FILENAME, PYEXEC_FLAG_SOURCE_IS_FD

Reply:
ok